### PR TITLE
`RightSidebar` improvements and bug fixes (participant tabs)

### LINKED ISSUE #9298: Custom permission flag shown for moderators
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Invite a user to a room
2. Grant "all permissions"
3. Promote to moderator

### Expected behaviour
Looks like a normal moderator

### Actual behaviour
Is a moderator but also shows custom permissions icon, but there is no way to reset:
image

### LINKED ISSUE #9345: Shared items tab focused for group chats
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

## Steps to reproduce
1. Open a group conversation
2. Make sure right sidebar is open and in Participants tab
3. Change to a 1-1 conversation
4. See the sidebar is closed
5. GO back to Public/group conversation

### Expected behaviour
Participants tab is shown

### Actual behaviour
Shared items tab is shown

## Talk app

**Talk app version:** master and 16.0.3

### PR DESCRIPTION
**Pull request by Antreesy**
### ☑️ Resolves

* Don't wait for tabs to be mounted in `$nextTick`
* Requires https://github.com/nextcloud/nextcloud-vue/pull/4004
* Fix #9345
---
* Change icon for moderators with custom permissions
* Fix #9298 (should be also reset to default on API side?)
---
* Check if participant belongs to conversation when updating his status (fix error when trying to get info from server about status while switching between conversations)

### 🚧 Tasks

- [ ] Code review

### 🏁 Checklist

- [x] ⛑️ Tests (unit and/or integration) are included or not required
- [x] 📘 API documentation in `docs/` has been updated or is not required
- [x] 📗 User documentation in https://github.com/nextcloud/documentation/tree/master/user_manual/talk has been updated or is not required
- [x] 🔖 Capability is added or not needed

### CODE REVIEW

File: src/components/RightSidebar/RightSidebar.vue [hunk 1]
```diff
@@ -273,30 +273,33 @@ export default {
 
 	watch: {
 		conversation(newConversation, oldConversation) {
-			if (!this.isRenamingConversation) {
-				this.conversationName = this.conversation.displayName
-			}
+			// Waiting for chat tab to mount / destroy
+			this.$nextTick(() => {
```
Comment (ShGKme) [inline src/components/RightSidebar/RightSidebar.vue:277]: It is not clean to me, why had we to wait until the chat tab is mounted or destroyed for all this actions?

Comment (Antreesy) [reply src/components/RightSidebar/RightSidebar.vue:277]: See watchers and methods for https://github.com/nextcloud/nextcloud-vue/blob/master/src/components/NcAppSidebar/NcAppSidebarTabs.vue:

If there is no tab provided (before `$nextTick`, for example), library component will switch to the first available tab ('shared items' in our case). So we need to mount 'participants' tab component, and only after that set `:active` to 'participants'

Comment (ShGKme) [reply src/components/RightSidebar/RightSidebar.vue:277]: The problem is in `NcAppSidebarTabs`, it ignores activating the current tab after registration.

Comment (Antreesy) [reply src/components/RightSidebar/RightSidebar.vue]: Works perfect with mentioned patch in vue-lib!

**Review by ShGKme — APPROVED** [review]
👀

**Comment by Antreesy** [comment]
Rebased onto master with @nextcloud/vue 7.11.1

See last commit: component styles was adjusted

File: src/components/RightSidebar/RightSidebar.vue [hunk 2]
```diff
@@ -375,6 +375,20 @@ export default {
...
+
+	.checkbox-radio-switch__label {
+		text-align: center;
+		justify-content: flex-start;
+	}
+
+	.checkbox-radio-switch__icon {
+		flex-basis: auto;
+	}
+}
```
Comment (ShGKme) [inline src/components/RightSidebar/RightSidebar.vue]: If I'm not wrong, `checkbox-radio-switch__label` and `checkbox-radio-switch__icon` are not in deep.

Comment (Antreesy) [reply src/components/RightSidebar/RightSidebar.vue]: Somehow they work as-is and don't work inside :deep()

**Review by ShGKme — APPROVED** [review]

**Pull request merged**